content/news/posts/2025/breaking-change-guzzle-removal-v9.md

```diff
@@ -0,0 +1,69 @@
+
+But this library has been replaced over the years by other tools in the core, most the Symfony HTTP Client. The core itself no longer had needs for this dependency, so we decided it was time to make the core lighter and remove guzzle along with other unused libraries.
+
+The weight of the build is not the only reason, in PrestaShop 8 we had to update Guzzle from version 5 to 7, this already created some incompatibilities at the time because of the breaking changes from Guzzle itself and modules struggled to maintain a cross compatibility between PrestaShop versions.
+One of the solution was to forcefully include Guzzle in the module dependencies, but when two modules do this with different Guzzle versions it creates some conflicts, and of course it could also have conflicts with the version from the core.
+
+To make things clear, at least from the core perspective, we made the drastic decision to remove Guzzle completely from our dependencies because we no longer use it so there's no point in integrating it and to prevent a nightmare of conflicts between the core and the modules.
+
+## How can I adapt my module?
+
+There are multiple solutions to overcome this missing library, here are a few examples.
+
+### Use Symfony HTTP Client (recommended)
+
+The best way to avoid conflicts is not to add a custom library and rely on the core ones as much as possible. As such Symfony is the core framework of PrestaShop and will remain it in the future, so it's a safe dependency that is sure never to disappear.
+It includes an HTTP client component which has been integrated in PrestaShop since the version 1.7.7 so if your module only targets this version and upper you can safely use the Symfony HTTP client, you can find details on how to use it on the [Symfony documentation page](https://symfony.com/doc/current/http_client.html).
+
+```php
+use Symfony\Component\HttpClient\HttpClient;
+
+$client = HttpClient::create();
+$response = $client->request('GET', 'https://api.domain.com');
+```
+
+### Use a custom HTTP client implementation
+
+You can create your own HTTP client, the advantage is that you are sure never to have conflicts with other modules that would use different library versions.
+
+The drawback is that you need to maintain it yourself and you re-invent the wheel.
+
+Here's an example of the client used in the `PrestaShop EventBus` module https://github.com/PrestaShopCorp/ps_eventbus/blob/27ca0e360d88345e57e945ad6369a215f4e615d0/src/Api/HttpClient.php
+
+### Forcefully include Guzzle in my module's vendors (not recommended)
+
+That's the quick and dirty solution, this way you don't need to adapt the code. But for reasons mentioned earlier we highly discourage it as your module is likely not to work with other modules because of potential conflicts.
+
+If you develop a specific module for a specific shop with a restricted list of modules it may be a quick solution, but if you sell your modules for multiple shops do not use this solution.
```

Comment:
```suggestion
If you develop a specific module for a specific shop with a restricted list of modules it may be a quick solution, but if you sell your modules for multiple shops, do not use this solution.
```